ghu sensing upstream pr3

ghu0510 opened the pull request:
sensing: hinge angle sensor initial
sensing: loop sensors in runtime module
sensing: implement sensor process data function
sensing: implement sleep time calculation for sensor loop
sensing: implement sensor post data processing

ghu0510 committed:
sensing: add hinge angle sensor skeleton

Add hinge angle virtual sensor for sensing subsystem, hinge angle sensor
takes both base accel and lid accel as reporter

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: implement set/get interval and sensitivity

set/get interval sensitivity is implmented in sensor management module,
and also add related test case

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: create runtime thread and implement later config

create runtime thread and wait event semphore, meanwhile, trigger event
semphore in set interval/sensitivity API to notify runtime thread to
arbitrate interval/sensitivity

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

yperess commented:
During the review of https://github.com/zephyrproject-rtos/zephyr/pull/55389 it was agreed to unblock it as long as design concerns are addressed before any additional changes are built on top of the subsystem. Please address https://github.com/zephyrproject-rtos/zephyr/issues/58925 first.

ghu0510 committed:
sensing: refine sensing API

1) modify reporter type from int to const sensing_sensor_handle_t in
sensing_sensor_process_t callback
2) move variable shift behind struct sensing_sensor_value_header in
struct sensing_sensor_value_q31
3) add pointer checking for sensing API

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: hinge angle sensor initial

implement hinge angle sensor init and set interval function

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: loop sensors in runtime module

add runtime module, implement loop sensors in runtime module

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: implement sensor process data function

1) enhance sensor need execute function, and data ready and sensor has
new data checking
2) implement sensor process data function, included both physical sensor
and virtual sensor data processing
3) implement sensor send dato to client, including time and sensitivity
checking and update sensor consume time

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: implement sleep time calculation for sensor loop

implement sleep time calculation in loop sensors

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

ghu0510 committed:
sensing: implement sensor post data processing

1) enhance sensor data process, check time, sensitivity condition
2) send data to its client
3) create dispatch thread to process sensor data from runtime module

Signed-off-by: Guangfu Hu <guangfu.hu@intel.com>

lixuzha commented:
Basing on design changed, submit https://github.com/zephyrproject-rtos/zephyr/pull/64478 to instead of this one. Closed this one.
